ast/desugar/Desugar.cc

```diff
@@ -2066,8 +2053,7 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
             [&](parser::CaseMatch *caseMatch) {
                 // Create a local var to store the expression used in each match clause
                 auto exprLoc = caseMatch->expr->loc;
-                auto exprName = dctx.ctx.state.freshNameUnique(core::UniqueNameKind::Desugar, core::Names::assignTemp(),
-                                                               dctx.uniqueCounter++);
```

Comment:
This is the lone call that was different, and whose change is responsible for the exp changes.